### PR TITLE
[batch] fix user cancel/delete request blocking if driver down

### PR DESCRIPTION
**Pull request by cseed**
This should put the issue of non-responsive cancels to rest.

### CODE REVIEW

File: batch/batch/front_end/front_end.py [hunk 1]
```diff
@@ -1034,6 +1019,30 @@
...
 
+async def cancel_batch_loop_body(app):
+    async with aiohttp.ClientSession(
+            raise_for_status=True, timeout=aiohttp.ClientTimeout(total=5)) as session:
+        await request_retry_transient_errors(
+            session, 'POST',
+            deploy_config.url('batch-driver', f'/api/v1alpha/batches/cancel'),
+            headers=app['driver_headers'])
+
+    # wait
```
Comment (jigold) [inline batch/batch/front_end/front_end.py]: What are the wait comments for?

Comment (cseed) [reply batch/batch/front_end/front_end.py]: To remind the reader that the return value indicates that the run_if_changed logic should wait.

I've forgotten the return value in the past, so I just wanted to emphasize it.

What is really want is named return values so I can document the return value by giving it the name the caller expects, something like: `return should_wait=True` which would correspond to this line in run_if_changed:

 >         should_wait = await f(*args, **kwargs)

but of course Python doesn't let you do that.

Comment (cseed) [reply batch/batch/front_end/front_end.py]: I could change it to:

```
  should_wait = True
  return should_wait
```

File: hail/python/hailtop/utils/utils.py [hunk 2]
```diff
@@ -227,3 +229,31 @@ def is_transient_error(e):
...
                 log.info(msg(exc), exc_info=True)
         await sleep_and_backoff(delay)
+
+
+async def retry_long_running(name, f, *args, **kwargs):
+    delay_secs = 0.1
+    while True:
+        try:
+            start_time = time_msecs()
+            await f(*args, **kwargs)
```
Comment (jigold) [inline hail/python/hailtop/utils/utils.py]: do you need a return here?

Comment (cseed) [reply hail/python/hailtop/utils/utils.py]: Hmm.  Maybe.  This is to retry long running things, and by long running, I mean things that we expect will never return.  But I guess they return, we should to.  Yes, I'll add it.  Good suggestion.

File: batch/batch/front_end/front_end.py (hunk 1 above)
Comment (cseed) [reply batch/batch/front_end/front_end.py]: I changed this to be the documentation-through-code style.

**Pull request merged**